### ISSUE TITLE
.ddg.txt.increment never called?

### ISSUE THREAD

**Issue by tfjmp** [End-to-end-provenance/RDataTracker#154]
This function fills the increment buffer:
.ddg.append.inc <- function(...) {
  increment <- .ddg.get("ddg.increment")
  .ddg.set("ddg.increment", paste(increment, ..., sep=""))
}

This function empty the buffer and return the value that was buffered:
.ddg.txt.increment <- function() {
  # Get current increment.
  dtxt <- .ddg.get("ddg.increment")
  # Reset for next increment.
  .ddg.set("ddg.increment", "")
  return(dtxt)
}

PS: at the moment that has the side effect of doubling memory consumption as ddg.txt and ddg.increment are identical.

We seems to have the architecture to increment the file and/or drawing, but it is not being used. And instead the whole file is written and/or redrawn every time. Is that correct? Do we want to get back and look at how increment is being implemented.

Can we discuss this at the next meeting?

**Comment by erboose** [comment]
Right, various strategies emerged in recent months for saving the provenance graph and these need to be sorted out.  And Moe's incremental drawing code (which uses the incremental data) needs to be checked and merged into the development branch.

We might also give some thought to handling and storing larger graphs.  An earlier version of RDataTracker, for example, appended the ddg.txt string to the ddg.txt file and cleared the string after every 100 nodes and edges were created.  This removed the need to handle a large string in memory and improved performance significantly for larger graphs.

**Comment by tfjmp** [comment]
@erboose we looked at the `development` branch and `incrementalDrawing`, is there another branch we should have looked at?

I am very happy for those things to be discussed during the next meeting.

**Comment by erboose** [comment]
Those two branches should cover current options for saving the graph.  It might be good to merge Barb's recent changes in CommandClass (a major update) with development before making further changes.

**Comment by MKLau** [comment]
Thanks Thomas and Emery. 

To expand on what prompted Thomas' and my looking into this, based on Thomas' demo of sharing DDG's using camflow, I asked him if it would be possible to get something like that running on the server where I'm going to be processing a bunch of sequence data very soon. It would be very helpful to be able to view the DDG as the process is running. Currently Camflow is still unstable, so we started looking into the incremental drawing in RDataTracker to see if it would do the job. 

As Thomas stated in the initial post, from what we could see, RDataTracker (including the incremental drawing functionality, for which Moe's documentation was very useful in understanding) is not building the DDG incrementally, though it looks like we're really close to implementing that. 

Per your suggestion, Emery, of merging those additions, I just forked the repo to test merging many of these features that are separated onto different branches. I'll report back with what I find.
